scripts/dependency_updater.py

```diff
@@ -0,0 +1,536 @@
+    for release in releases:
+        version = release.get('tagName', '')
+        if stable_version_pattern.match(version):
+            patch_versions.append(version)
+    # Fallback to tags
+    if not patch_versions:
+        tags = component_repo_metadata.get('refs', {}).get('nodes', [])
+        for tag in tags:
+            version = tag.get('name', '')
+            if stable_version_pattern.match(version):
+                patch_versions.append(version)
+    patch_versions.sort(key=lambda v: list(map(int, re.findall(r'\d+', v)))) # sort for checksum update
+    return patch_versions
+
+def get_repository_metadata(component_info, session):
+    query_parts = []
+    for component, data in component_info.items():
+        owner = data['owner']
+        repo = data['repo']
+        query_parts.append(f"""
+            {component}: repository(owner: "{owner}", name: "{repo}") {{
+                releases(first: {args.graphql_number_of_entries}, orderBy: {{field: CREATED_AT, direction: DESC}}) {{
+                    nodes {{
+                        tagName
+                        url
+                        description
+                        publishedAt
+                        isLatest
+                    }}
+                }}
+                refs(refPrefix: "refs/tags/", first: {args.graphql_number_of_entries}, orderBy: {{field: TAG_COMMIT_DATE, direction: DESC}}) {{
+                    nodes {{
+                        name
+                        target {{
+                            ... on Tag {{
+                                target {{
+                                    ... on Commit {{
+                                        history(first: {args.graphql_number_of_commits}) {{
+                                            edges {{
+                                                node {{
+                                                    oid
+                                                    message
+                                                    url
+                                                }}
+                                            }}
+                                        }}
+                                    }}
+                                }}
+                            }}
+                            ... on Commit {{
+                                # In case the tag directly points to a commit
+                                history(first: {args.graphql_number_of_commits}) {{
+                                    edges {{
+                                        node {{
+                                            oid
+                                            message
+                                            url
+                                        }}
+                                    }}
+                                }}
+                            }}
+                        }}
+                    }}
+                }}
+            }}
+        """)
+
```

Comment:
On the query.
First, you should really put it in a separate file, so it can be edited and run separately.
Secondly, you should use graphql variables instead of templating the query and templating and joining.
For instance something like this :
```graphql
query($repoWithReleases: [ID!]!, $repoWithTags: [ID!]!) {
  with_releases: nodes(ids: $repoWithReleases) {

    ... on Repository {
      nameWithOwner
      releases(first: 100) {
        nodes {
          tagName
          isPrerelease
          releaseAssets {
            totalCount
          }
        }
      }
    }
  }

  with_tags: nodes(ids: $repoWithTags) {

    ... on Repository {
      nameWithOwner
      refs(refPrefix: "refs/tags/", last: 100) {
        nodes {
          name
        }
      }
    }
  }
}
```